SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -38,11 +39,29 @@ public IEnumerable<CareCaseData> GetProcessData(ListCasesRequest request)
                     throw new Exception();
                 }
             }
-            var filter = !string.IsNullOrEmpty(request.WorkerEmail) ?
-                Builders<BsonDocument>.Filter.Eq("worker_email", request.WorkerEmail)
-                : "{$or: [{ mosaic_id: " + mosaicId.ToString() + "}, { mosaic_id: /" + mosaicId.ToString() + "/}]}";
+            var officerEmailFilter = !string.IsNullOrWhiteSpace(request.WorkerEmail) ?
+                Builders<BsonDocument>.Filter.Eq("worker_email", request.WorkerEmail) : null;
+            var mosaicIdFilter = !string.IsNullOrWhiteSpace(mosaicId.ToString()) ?
+                Builders<BsonDocument>.Filter.Eq("mosaic_id", mosaicId.ToString()) : null;
+            var firstNameFilter = !string.IsNullOrWhiteSpace(request.FirstName) ?
+                Builders<BsonDocument>.Filter.Regex("first_name".ToUpper(), request.FirstName.ToUpper()) : null;
+            var lastNameFilter = !string.IsNullOrWhiteSpace(request.LastName) ?
+                Builders<BsonDocument>.Filter.Regex("last_name".ToUpper(), request.LastName.ToUpper()) : null;
+            var dateOfBirthFilter = !string.IsNullOrWhiteSpace(request.DateOfBirth.ToString()) ?
```

Comment:
DateOfBirth is already string so perhaps no need for the conversion here

Reply:
Also spotted that, thank you!